### PR TITLE
Add a 'multi' transport

### PR DESCRIPTION
**Pull request by freman**
Permit trying multiple transports in a given order.

eg:
```go
    c := client.NewClient(
        client.Transport(
            multi.NewTransport(
                multi.WithTransports(
                    transport.NewTransport(transport.Secure(true)),
		    transport.NewTransport(transport.Secure(false)),
                ),
            ),
        ),
    )
```

### CODE REVIEW

File: transport/multi/multi.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+type multi struct {
+	transports []transport.Transport
+	listen     transport.Transport
+}
+
+func (m *multi) Dial(addr string, opts ...transport.DialOption) (client transport.Client, err error) {
+	err = errors.New("No transports provided")
+
+	for _, transport := range m.transports {
```
Comment (asim) [inline transport/multi/multi.go]: Should this do more? If it's a multi transport should it perhaps look at "addr" and dial based on some prefix e.g `tcp://localhost:6060`.

```
parts := strings.Split("://", addr)
if len(parts) > 1 {
  if t, ok := transports[parts[0]]; ok {
    return t.Dial(parts[1], opts...)
  } 
}
```

Just a thought

File: transport/multi/multi.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	m := &multi{}
+	for _, o := range opts {
+		o(m)
+	}
+	return m
+}
+
+// WithTransports allows you to specify which transports in the order
+// of use that you wish to have them attempted.
+func WithTransports(transport ...transport.Transport) Option {
```
Comment (asim) [inline transport/multi/multi.go]: Call it WithTransport

File: transport/multi/multi.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	return func(m *multi) {
+		m.transports = append(m.transports, transport...)
+	}
+}
+
+// WithListenTransport allows you to specify an optional listening transport.
+func WithListenTransport(transport transport.Transport) Option {
```
Comment (asim) [inline transport/multi/multi.go]: WithListener

File: transport/multi/multi.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+	}
+	return nil, errors.New("Not supported")
+}
+
+func (m *multi) String() string {
+	return "multi"
+}
+
+// NewTransport will return a new multi Transport with the given options.
+func NewTransport(opts ...Option) transport.Transport {
```
[the remark this answers is not in the document]
Comment (freman) [reply transport/multi/multi.go]: Fair enough, I was looking too much at http_transport.go when I wrote this, not at the go-plugins/transports

**Comment by freman** [comment]
not doing it this way any more.

**Pull request closed without merging**